Matching for wildcards in topics

eranor opened the pull request:
Hi, @davidyaha. 
I have added a filter to check if any of the subscriptions match the message topic, while matching on wildcards too. Currently on message receive it does not find a key for `topic/a`, if one subscribes for `topic/#` because it only checks if the topic exists as a key in the subscriptions dictionary.

In our use case we would like to allow us to set the topic more generally without specifying the whole path.

eranor committed:
Added matching for wildcards in topics

eranor committed:
Removed unused variable

davidyaha commented:
Hey @eranor! Thanks for taking interest in this project!

Would you mind specifying what this PR is for?
How it will be used?
Also could you add tests for this feature? 

Thanks ahead!

eranor committed:
Added tests

eranor commented:
Added description and the tests requested

davidyaha commented:
@eranor Thanks for updating this PR!
This actually a great addition to support the mqtt wildcards!
It is currently not passing the CI tests due to TS errors, have you seen those while running locally?

Thanks again!

eranor committed:
Fix TS type errors

eranor committed:
Replace unsupported function Object.entries

eranor committed:
Replace Array.includes with Array.indexOf

eranor committed:
Minor refactoring

Re-order method and field declarations


Add missing curly brackets


Replace operator at indexOf check

eranor commented:
@davidyaha Hi. I've corrected the Typescript and other errors causing the CI to fail.

davidyaha commented:
Merged! Thanks!
